torch/fx/experimental/proxy_tensor.py

```diff
@@ -414,7 +414,8 @@ def can_handle_tensor(x):
     else:
         constant = None
 
-    track_tensor_tree(out, proxy_out, constant=constant, tracer=tracer)
+    with inside_mode(proxy_mode):
```

Comment:
This seems surprising to me.

Reply:
still waiting on explainer

Reply:
sigh, wrote the comment when I initially made this update and forgot to submit it.

I'd be happy to make this a separate PR. The issue was that:

(1) We detach tensors as part of creating proxies.

(2) In pre_autograd tracing, we currently push `TorchProxyDispatchMode` onto **both** the autograd mode stack, and the original python key mode stack

(3) The detach() calls get intercepted by the proxy mode on the original python key mode stack.

So we need to be careful that any aten ops that we call inside of TorchProxyDispatchMode happen in a `with inside_mode()` now.

I think we agreed 2-3 weeks ago that we could avoid re-entrant issues if we solved the "fallthrough keys can't be intercepted by the python dispatcher" issue, but we agreed this is difficult.

Reply:
Ok put this comment in the code?